### DISCUSSION TITLE
Can we increase the workflows_pool_size beyond 40?

### DISCUSSION THREAD

**Discussion by zsmanjot** [StackStorm/st2#5652] [Workflows]
We are working on deploying our workflow and saw that the system got slow when more than 300 workflows were put up in delayed queue depending upon the concurrency policy that we have set. 

Can we have a provision where we can change this hard limit beyond 40 as we have a lot of nodes that we target at a time and it just gets slow at some point of time.

**Comment by mgazzali** [comment]
Assuming that you install st2 using Kubernetes you will have a configmap that used for the st2 component.

There is one configmap called "st2-config" that able to modify the poll size under st2scheduler 

```
[scheduler]
# How long GC to search back in minutes for orphaned scheduled actions
execution_scheduling_timeout_threshold_min = 1
# How often (in seconds) to look for zombie execution requests before rescheduling them.
gc_interval = 3
# Location of the logging configuration file.
logging = /etc/st2/logging.scheduler.conf
# The size of the pool used by the scheduler for scheduling executions.
pool_size = 100
# The maximum number of attempts that the scheduler retries on error.
retry_max_attempt = 10
# The number of milliseconds to wait in between retries.
retry_wait_msec = 3000
# How long (in seconds) to sleep between each action scheduler main loop run interval.
sleep_interval = 0.1
```
